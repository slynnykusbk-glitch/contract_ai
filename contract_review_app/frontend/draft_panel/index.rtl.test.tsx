import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

const meta = { rules_evaluated: 1, rules_triggered: 0 };

jest.mock('../common/http', () => ({
  postJSON: jest.fn(),
  getHealth: jest.fn().mockResolvedValue(meta),
  ensureHeadersSet: jest.fn(),
}));

import { DraftAssistantPanel } from './index';

test('renders placeholder when findings is null', async () => {
  const analysis = { findings: null };
  const { container } = render(
    <DraftAssistantPanel initialAnalysis={analysis} initialMeta={meta} />
  );
  await screen.findByText(
    'No findings (rules_evaluated: 1, triggered: 0)'
  );
  expect(screen.queryByText(/^Error:/)).toBeNull();
  expect({ analysis, meta }).toMatchSnapshot();
});
