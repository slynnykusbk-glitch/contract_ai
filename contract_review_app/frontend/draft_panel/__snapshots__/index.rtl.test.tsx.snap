// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders placeholder when findings is null 1`] = `
{
  "analysis": {
    "findings": null,
  },
  "meta": {
    "rules_evaluated": 1,
    "rules_triggered": 0,
  },
}
`;
